.github/workflows/ansible-linter.yaml

```
@@ -0,0 +1,22 @@
+name: ansible-lint
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+
+jobs:
+  build:
+    name: Ansible Lint # Naming the build is important to use it as a status check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run ansible-lint
+        uses: ansible/ansible-lint@main # or version tag instead of 'main'
+      - name: Upload sarif
+        if: always()
```

Could you explain the use of the `always()` function here?

The [documentation](https://docs.github.com/en/actions/learn-github-actions/expressions#always) explains clearly what it is meant to do, but I do not understand why it should be needed here.

Also, from reading the example in the documentation, is this syntactically correct? Shouldn't it be like this: 
```suggestion
        if: {{ always() }}
```

---

We need it because the ansible-lint step will fail if the lint step fails. If a step fails, it will automatically skip all the next steps, so we force the action to run the upload SARIF step even if the previous step failed (to get the code annotations automatically and to comment on what needs to be changed).

The action is syntactically correct because Github Action implicitly adds the `${{}}` delimiter AFAIK

---

I see, got it.